src/components/SegmentedTopBar/index.jsx

```diff
@@ -0,0 +1,130 @@
+import React, {
+  createRef,
+  useState,
+  useMemo,
+  useRef,
+  useLayoutEffect,
+} from "react";
+import { StyleSheet, View } from "react-native";
+
+import PropTypes from "prop-types";
+import { moderateScale } from "react-native-size-matters";
+
+import { Container } from "@components";
+
+import { Indicator } from "./Indicator";
+import { Tab } from "./Tab";
+
+/**
+ *
+ * A Segment Picker component that provides toggling between two or more views.
+ *
+ * <div class="screenshots">
+ *   <img src="screenshots/toptabbar/toptabbar-1.png" />
+ *   <img src="screenshots/toptabbar/toptabbar-2.png" />
+ * </div>
+ *
+ *   ## Usage
+ * ```js
+ * import * as React from 'react';
+ * import { createMaterialTopTabNavigator } from "@react-navigation/material-top-tabs";
+ * import { NavigationContainer } from "@react-navigation/native";
+ *
+ * import { SegmentedTopBar } from '@bigbinary/neetoui-rn';
+ *
+ * const Tab = createMaterialTopTabNavigator();
+ *
+ * export default function Main(){
+ *
+ *  return (
+ *    <NavigationContainer>
+ *      <Tab.Navigator tabBar={props => <SegmentedTopBar {...props} />} >
+ *        <Tab.Screen
+ *          name="Home"
+ *          component={HomeScreen}
+ *        />
+ *
+ *        <Tab.Screen
+ *          name="Settings"
+ *          component={SettingsScreen}
+ *        />
+ *      </Tab.Navigator>
+ *    </NavigationContainer>
+ *  );
+ * }
+ * ```
+ */
+
+export const SegmentedTopBar = ({
+  state: { routes, index },
+  navigation,
+  height,
+  descriptors,
+}) => {
+  const containerRef = useRef();
+
+  const [measures, setMeasures] = useState([]);
+
+  const tabsData = useMemo(
+    () =>
+      routes.map(({ key, name }) => {
+        const { options } = descriptors[key];
+        const label = options.tabBarLabel ?? options.title ?? name;
+
+        return { label, value: name, ref: createRef() };
+      }),
+    []
+  );
+
+  useLayoutEffect(() => {
+    const _measures = [];
+
+    tabsData.forEach(({ ref }) => {
+      ref.current.measureLayout(containerRef.current, (x, y, width, height) => {
+        _measures.push({ x, y, width, height });
+
+        if (_measures.length === tabsData.length) {
+          setMeasures(_measures);
+        }
+      });
+    });
+  }, [measures.length]);
+
+  return (
+    <Container
+      bg="background.secondary"
+      borderRadius={moderateScale(6)}
+      mx={moderateScale(16)}
```

Comment:
@chiragbhaiji  Looks like the width is not 100% by default. Need to remove this margin horizontal.

<img width="576" alt="Screenshot 2023-03-25 at 9 42 05 AM" src="https://user-images.githubusercontent.com/93119254/227695252-0c8f3a1d-84ef-474a-874f-b940442c7c90.png">